test/generateTypes-group.test.ts

```diff
@@ -56,6 +56,51 @@ it("creates an interface for a group item containing its fields", (ctx) => {
 	).toBe("prismic.SelectField");
 });
 
+it("supports nested groups", (ctx) => {
+	const model = ctx.mock.model.customType({
+		id: "foo",
+		fields: {
+			bar: ctx.mock.model.group({
+				fields: {
+					baz: ctx.mock.model.group({
+						fields: {
+							qux: ctx.mock.model.keyText(),
+							quux: ctx.mock.model.select(),
+						},
+					}),
+				},
+			}),
+		},
+	});
+
+	const types = lib.generateTypes({ customTypeModels: [model] });
+	const file = parseSourceFile(types);
+
+	const nestedGroupProperty = file
+		.getInterfaceOrThrow("FooDocumentDataBarItem")
+		.getPropertyOrThrow("baz");
+	expect(nestedGroupProperty.getTypeNodeOrThrow().getText()).toBe(
+		"prismic.NestedGroupField<Simplify<FooDocumentDataBazItem>>",
+	);
```

Comment:
⚠️ #issue: I think this creates potential collisions if you have another group field named `baz` at the root of your document or another nested group field named `baz` elsewhere (but correct me if I got it terribly wrong).

I believe this could happen quite often with nested groups named `buttons`, `ctas`, `images`, and such.

Not sure what naming scheme we should follow however to prevent all kind of collisions, as naming the nested group interface in such fashion `FooDocumentDataBarItemBazItem` would then make it collide with a group named `BarItemBaz` (unlikely for sure, maybe we're fine with it)

I see some possible naming like `FooDocumentData_Bar_BazItem` (leading `_` preventing collisions), but I'm not such I like them. Maybe you'll come up with something more clever? Happy to discuss about it :)

Reply:
Good catch! I think we can go with this naming scheme: `FooDocumentDataBarBazItem`, where `Bar` is the parent group and `Baz` is the child group.

I implemented that method here: https://github.com/prismicio/prismic-ts-codegen/pull/63/commits/85ebe0252f7a9bcbcb5c52785d5785fe032ac46c

It's possible to get a name collision when a model has the following fields:
- Has a `bar` group.
- Has a `baz` group within `bar`.
- Has a `bar_baz` group.

But I think it's okay. That situation is unlikely, and devs can rename their fields accordingly.

If it becomes an issue, we can write something that automatically suffixes type names with a counter (e.g. `Foo`, `Foo2`, `Foo3`).

Reply:
Works for me!